README.md

```diff
@@ -43,30 +43,61 @@ $ git clone https://github.com/iterative/example-gto.git
 $ cd example-gto
 ```
 
-### Versioning
+This guide will walk you through the Lifecycle of an artifact. To execute each
+step, you'll need to create a Git tag of a special format, which will trigger CI
+and trigger the job you intend to run:
 
-To register a new artifact or a new version, use `gto register`. This is usually
-done to mark significant changes to the artifact (such as a release or a
-deprecation).
+1. Registering artifact - marking the very beginning of the artifact history in
+   the Git repository.
+2. Publishing a version - marking an important change in the artifact.
+3. Assigning a stage to a version - marking the readiness to be consumed by a
+   downstream system.
+4. Unassigning a stage - removing an "artifact is ready for this stage" mark.
+5. Deprecating a version - marking a version as the one that should not be used
+   any longer.
+6. Deregistering an artifact - completing the lifecycle of an artifact and
+   marking it as an outdated one.
+
+### Registering an artifact
+
+Note: this is an optional operation, since publishing the very first version
+with `$ gto publish` already implies registering the artifact.
+
+To start tracking the artifact in the registry, you need to run:
 
 ```console
 $ gto register awesome-model
-Created git tag 'awesome-model@v0.0.1' that registers a new version
+Created Git tag `awesome-model@` that registers the artifact.
+```
+
+This can be done to mark the very beginning of the artifact history, so it's
+better to create this tag for the very first commit the artifact was present in
+(even not annotated - see below what an annotation is).
+
+### Versioning an artifact
+
+To release a new artifact or a new version, use `gto publish`. This is usually
```

Comment:
Why is this called `publish` instead of `version` or `register-version` or something like that?

Reply:
`register-version` is a bit long IMO. Could be `version`, what is the pair command? `unversion`?
Then it could be `stage` instead of `assign`. `unstage` sounds ok to me.

Reply:
`version` and `stage` sound good. But since these are words that used more often as nouns than as verbs, they may confuse users. I did suggest `version`, but may be `register-version` or `add-version` is clearer?

Reply:
Then it may be `$ gto publish-version` with an alias `$ gto publish`. Or `register-version` with alias `register`.